electron_app/DownloadManager.js

```diff
@@ -0,0 +1,55 @@
+const mtd = require('zeltice-mt-downloader');
+const path = require('path');
+const os = require('os');
+
+const handleFolder = require('./helpers/_handleDisk');
+
+let downElements = [];
+
+let downloadManager = (items, access_token) => {
+    let filePath = '';
+
+    let options = {
+        count: 8,
+        headers: {
+            'Authorization': 'Bearer ' + access_token,
+        },
+        onStart: function (meta) {
+            console.log('Download started');
+        },
+        onEnd: function (err, result) {
+            if (err) console.error(err);
+            else {
+                console.log('Download Complete');
+            }
+        }
+    };
+
+    let promiseArr = [];
+    items.forEach(item => {
+        if (item.preserveStructure) {
+            const structurePath = item.path.substring(item.path.lastIndexOf('/'), 0);
+            promiseArr.push(handleFolder(path.join(item.destination, structurePath), item));
+        } else {
+            promiseArr.push(new Promise((res, rej) => { res({ 'path': item.destination, 'item': item }); }));
+        }
+    });
+    Promise.all(promiseArr).then(result => handleDownloads(result, options));
```

Comment:
Is the code using promises here to avoid having all files fail if one file fails to create a directory?